float64 position1
float64 position2
float64 position3
float64 position4
